controller/utils.go

```diff
@@ -39,41 +39,59 @@ func replacePodsAndWait(
 	willTimeoutAt time.Time,
 	clock clockinterfaces.Clock,
 	configYAML *models.ConfigYAML,
-	podsToDelete []v1.Pod,
+	podsChunk []v1.Pod,
 	scheduler *models.Scheduler,
 	operationManager *models.OperationManager,
 ) (createdPods []v1.Pod, deletedPods []v1.Pod, timedout, canceled bool) {
 	createdPods = []v1.Pod{}
 	deletedPods = []v1.Pod{}
 
-	for _, pod := range podsToDelete {
-		logger.Debugf("deleting pod %s", pod.GetName())
+	// create a chunk of pods (chunkSize = maxSurge)
+	for _, pod := range podsChunk {
```

Comment:
This way I think maxSurge is locked to 1 no matter what. What needs to be done is invoking the block inside this for in routines and waiting on all of them (using channel or sync.WaitGroup) to finish before continuing the loop

Reply:
What you're suggesting is for each pod in the chunk to be created in parallel? If so, we could have the problem of creating too many pods at once